cmd/kubeadm/app/cmd/upgrade/node.go

```diff
@@ -104,28 +103,32 @@ func newCmdNode(out io.Writer) *cobra.Command {
 	return cmd
 }
 
-// newNodeOptions returns a struct ready for being used for creating cmd kubeadm upgrade node flags.
-func newNodeOptions() *nodeOptions {
-	kubeConfigPath := constants.GetKubeletKubeConfigPath()
-
-	// isControlPlaneNode checks if a node is a control-plane node by looking up
-	// the kube-apiserver manifest file
+// isControlPlaneNode checks if a node is a control-plane node by looking up
+// the kube-apiserver manifest file
+func isControlPlaneNode() bool {
 	isControlPlaneNode := true
 	filepath := constants.GetStaticPodFilepath(constants.KubeAPIServer, constants.GetStaticPodDirectory())
 	if _, err := os.Stat(filepath); os.IsNotExist(err) {
 		isControlPlaneNode = false
 	}
+	return isControlPlaneNode
+}
 
-	if isControlPlaneNode {
+// newNodeOptions returns a struct ready for being used for creating cmd kubeadm upgrade node flags.
+func newNodeOptions() *nodeOptions {
+	kubeConfigPath := constants.GetKubeletKubeConfigPath()
+	if isControlPlaneNode() {
```

Comment:
Thank you for your quick response @neolit123 
I'm afraid the result of `isControlPlaneNode()` is always false here when the chroot is not invoked.
Maybe we should leave the default value of `--kubeconfig` flag as empty, and if the user does not set it, it will be filled automatically when `newNodeData` is executed according to the result of `isControlPlaneNode()`.

Reply:
> I'm afraid the result of isControlPlaneNode() is always false here when the chroot is not invoked.

yes, it's a strange case, but i thought that it's mostly fine because the --rootfs is not a common use case.
however i agree the discrepancy on flag default value is not great.
FWIW, AddKubeConfigFlag overrides the default value to admin.conf, so that was already an existing behavior.

> Maybe we should leave the default value of --kubeconfig flag as empty, and if the user does not set it, it will be filled automatically when newNodeData is executed according to the result of isControlPlaneNode().

updated to do that.
PTAL.